pkg/daemon/sync_test.go

```diff
@@ -406,8 +406,126 @@ func TestDoSync_WithNewCommit(t *testing.T) {
 	}
 }
 
+func TestDoSync_WithKustomize(t *testing.T) {
+	d, cleanup := daemon(t, testfiles.FilesForKustomize)
+	defer cleanup()
+
+	d.GitConfig.Paths = []string{"staging"}
+	d.ManifestGenerationEnabled = true
+
+	ctx := context.Background()
+
+	var syncTag = "syncy-mcsyncface"
+	// Set the sync tag to head
+	var oldRevision, newRevision string
+	err := d.WithWorkingClone(ctx, func(checkout *git.Checkout) error {
+		ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+		defer cancel()
+
+		var err error
+		tagAction := git.TagAction{
+			Tag:      syncTag,
+			Revision: "master",
+			Message:  "Sync pointer",
+		}
+		err = checkout.MoveTagAndPush(ctx, tagAction)
+		if err != nil {
+			return err
+		}
+		oldRevision, err = checkout.HeadRevision(ctx)
+		if err != nil {
+			return err
+		}
+
+		// Push some new changes
+		absolutePath := path.Join(checkout.Dir(), "base", "foo.yaml")
+		def, err := ioutil.ReadFile(absolutePath)
+		if err != nil {
+			return err
+		}
+
+		newDef := bytes.Replace(def, []byte("key: value"), []byte("key: value2"), -1)
+		if err := ioutil.WriteFile(absolutePath, newDef, 0600); err != nil {
+			return err
+		}
+
+		commitAction := git.CommitAction{Author: "", Message: "test commit"}
+		err = checkout.CommitAndPush(ctx, commitAction, nil, true)
+		if err != nil {
+			return err
+		}
+		newRevision, err = checkout.HeadRevision(ctx)
+		return err
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = d.Repo.Refresh(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	syncCalled := 0
+	var syncDef *cluster.SyncSet
+	k8s.SyncFunc = func(def cluster.SyncSet) error {
+		syncCalled++
+		syncDef = &def
+		return nil
+	}
+
+	head, err := d.Repo.BranchHead(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gitSync, _ := fluxsync.NewGitTagSyncProvider(d.Repo, syncTag, "", fluxsync.VerifySignaturesModeNone, d.GitConfig)
+	syncState := &lastKnownSyncState{logger: d.Logger, state: gitSync}
+
+	if err := d.Sync(ctx, time.Now().UTC(), head, syncState); err != nil {
+		t.Error(err)
+	}
+
+	// It applies everything
+	if syncCalled != 1 {
+		t.Errorf("Sync was not called once, was called %d times", syncCalled)
+	} else if syncDef == nil {
+		t.Errorf("Sync was called with a nil syncDef")
+	}
+
+	// The emitted event has no workload ids
```

Comment:
This comment seems to be misleading -- what is tested below is that there's an event _with_ a workload ID

Reply:
Fixed. Thank you!